template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kafka/KafkaLdapConfigProviderTest.java

```diff
@@ -0,0 +1,130 @@
+        exp2.setName("ldap.auth.user.dn.template");
+        exp2.setValue("pattern");
+
+        ApiClusterTemplateConfig exp3 = new ApiClusterTemplateConfig();
+        exp3.setName("ldap.auth.enable");
+        exp3.setValue("true");
+
+        assertThat(serviceConfigs).hasSameElementsAs(List.of(exp1, exp2, exp3));
+    }
+
+    @Test
+    void isConfigurationNeeded() {
+        when(cmTemplateProcessor.isRoleTypePresentInService(anyString(), anyList())).thenReturn(true);
+        TemplatePreparationObject tpo = Builder.builder()
+                .withLdapConfig(LdapViewBuilder.aLdapView().build())
+                .build();
+        assertThat(underTest.isConfigurationNeeded(cmTemplateProcessor, tpo)).isTrue();
+    }
+
+    @Test
+    void isConfigurationNotNeeded() {
+        when(cmTemplateProcessor.isRoleTypePresentInService(anyString(), anyList())).thenReturn(false);
+        TemplatePreparationObject tpo = Builder.builder()
+                .withLdapConfig(LdapViewBuilder.aLdapView().build())
+                .build();
+        assertThat(underTest.isConfigurationNeeded(cmTemplateProcessor, tpo)).isFalse();
+    }
+
+    @Test
+    void isConfigurationNotNeeded2() {
+        lenient().when(cmTemplateProcessor.isRoleTypePresentInService(anyString(), anyList())).thenReturn(true);
+        TemplatePreparationObject tpo = Builder.builder()
+                .build();
+        assertThat(underTest.isConfigurationNeeded(cmTemplateProcessor, tpo)).isFalse();
+    }
+
+    @Test
+    void getRoleTypes() {
+        assertThat(underTest.getRoleTypes()).hasSameElementsAs(List.of("KAFKA_BROKER"));
+    }
+
+    @Test
+    void getServiceType() {
+        assertThat(underTest.getServiceType()).isEqualTo("KAFKA");
+    }
+
+    private TemplatePreparationObject getTemplatePreparationObject(String... locations) {
+        HostgroupView master = new HostgroupView("master", 1, InstanceGroupType.GATEWAY, 1);
+        HostgroupView worker = new HostgroupView("worker", 2, InstanceGroupType.CORE, 2);
+
+        List<StorageLocationView> storageLocations = new ArrayList<>();
+        if (locations.length >= 1) {
+            StorageLocation hmsExternalWarehouseDir = new StorageLocation();
+            hmsExternalWarehouseDir.setProperty("hive.metastore.warehouse.external.dir");
+            hmsExternalWarehouseDir.setValue(locations[0]);
+            storageLocations.add(new StorageLocationView(hmsExternalWarehouseDir));
+        }
+        S3FileSystemConfigurationsView fileSystemConfigurationsView =
+                new S3FileSystemConfigurationsView(new S3FileSystem(), storageLocations, false);
+
+        return TemplatePreparationObject.Builder.builder().withFileSystemConfigurationView(fileSystemConfigurationsView)
+                .withHostgroupViews(Set.of(master, worker)).build();
+    }
+
+    private String getBlueprintText(String path) {
+        return FileReaderUtils.readFileFromClasspathQuietly(path);
+    }
```

Comment:
these methods are not in use, kindly asking to remove them

Reply:
removed